index.js

```diff
@@ -18,20 +18,41 @@ module.exports = function (cmd, args, opts) {
 				val = stripEof(val);
 			}
 
+			if (parsed.options.encoding !== undefined) {
+				val = new Buffer(val, parsed.options.encoding);
+			}
+
 			return val;
 		};
 
-		var spawned = childProcess.execFile(parsed.command, parsed.args, parsed.options, function (err, stdout, stderr) {
```

Comment:
Using spawning misses out on a lot of the execFile stuff, like maxBuffer, which will need to be implemented manually, I think. https://github.com/nodejs/node/blob/cc82e5e3d9aefb5e31dbe688b0038327bb7d9b80/lib/child_process.js#L118-L300

Might be easier to just use spawn for the `.all` property.